src/core_test/mpas_test_core_field_tests.F

```diff
@@ -154,6 +155,52 @@ subroutine test_core_attribute_list_test(domain, threadErrs, ierr)!{{{
             call mpas_log_write('      Duplicate string does not match', MPAS_LOG_ERR)
          end if
 
+         allocate(modifyIntA(3))
+         allocate(modifyRealA(5))
+
+         modifyInt = 2
+         modifyIntA(:) = 2
+         modifyReal = 2.0_RKIND
+         modifyRealA(:) = 2.0_RKIND
+         modifyText = 'Modified'
+
+         call mpas_modify_att(srcList, 'testInt', modifyInt)
+         call mpas_modify_att(srcList, 'testIntA', modifyIntA)
+         call mpas_modify_att(srcList, 'testReal', modifyReal)
+         call mpas_modify_att(srcList, 'testRealA', modifyRealA)
+         call mpas_modify_att(srcList, 'testText', modifyText)
+
+         call mpas_get_att(srcList, 'testInt', destInt)
+         call mpas_get_att(srcList, 'testIntA', destIntA)
+         call mpas_get_att(srcList, 'testReal', destReal)
+         call mpas_get_att(srcList, 'testRealA', destRealA)
```

Comment:
It looks like the `mpas_get_att` routine allocates memory for the returned array in the case of a 1-d real or integer attribute, so after the previous set of tests (and after these tests), I think it may be necessary to deallocate `destIntA` and `destRealA` to avoid a memory leak.

Since this is just a testing core, a few small memory leaks may in practice be inconsequential, but working towards a situation in which we can run the entire test core with valgrind and not turn up any memory leaks would be nice, I think.